rfc-164-retire-email-alert-monitoring-as-a-separate-app.md

```diff
@@ -0,0 +1,100 @@
+
+### Problems with matching subject lines
+
+Problems can occur when trying to match by subject line if the title of the alert is altered after the email has been sent out. The [RSS feed] or [healthcheck URL] will contain a title that differs from the subject line of the email sent out. This will cause a failed match, and an alert will go off, even though for practical purposes the email has been sent. This causes false alarms, and needs someone to add a [hard-coded exception] into the matching code, increasing toil.
+
+[hard-coded exception]: https://github.com/alphagov/email-alert-monitoring/blob/79a865dcd8be07447735ae8ae99b78002241504a/lib/email_verifier.rb#L8-L36
+
+### Problems with matching by times
+
+Since the travel advice system matches by time and subject, this can lead to additionalo false alarms if the email being sent out is delayed over a minute boundary (since the code is matching on the minute the alert was created). Again, this has to be handled by a [hard-coded execption].
+
+[false alarms]: https://github.com/alphagov/email-alert-monitoring/pull/106
+
+### What the Notify callback can tell us about the status of an email
+
+The Notify callback gives us a [current reciept status] for an email, which is "delivered" if Notify receives a callback from Amazon SES (the backing service Notify uses to delivery emails) to confirm the mailserver handling emails for the recipient address has accepted the email. This is a reasonably solid assurance that the email has got to the recipient, and is definitely at the point where further handling of the email is outside of our control - the only things that might stop the recipient seeing the email at that point are:
+
+- The mailserver might fail to deliver internally: Valid, but we can't currently test that unless the failure occurs in the exact Gmail server handling our account anyway.
+- A spam filter might redirect the message: Untestable, relies on idiosyncratic account setup, and out of our control
+- The recipient might ignore the message: Untestable, and out of our control.
+
+### How we could replace this with a job internal to email-alert-api
+
+We currently keep a record of every single email sent out via email-alert-api (the Email model). These records are [kept for one week, then deleted]. This model could have a notify_status column added which begins nil and is then filled in by the callback. As long as one email sent in response to an alert has a notify_status of "delivered", we would have roughly the same assurance as we do now that the email alert has been sent successfully. It would also allow us the opportunity (if we wanted) to make a more stringent alert (say, that a certain percentage of the alert emails were delivered successfully) that we can't currently test.
```

Comment:
A positive use of keeping the email box is an email should always be attempted for each travel / drug alert, even if there are no actual subscribers. We may think about a mechanism that prevents the alert being fired if there are no subscribers to hopefully prevent false positives (e.g if for some reason we the our Gmail box doesn't receive the email). Seems low risk but possibly something to factor in.